specs/Standards/NonFungibleToken/Event.md

```diff
@@ -0,0 +1,101 @@
+# Non-Fungible Token Event
+
+Version `1.0.0`
+
+## Summary
+
+Standard interfaces for NFT contract actions.
+
+## Motivation
+
+NFT-driven apps such as marketplaces regularly perform many
+of the same actions such as `minting`, `burning` and `transferring`.
+Each app have their own way of performing these actions and it
+is difficult to consistently capture these events.
+This extension addresses that.
+
+For example, it's common in NFT marketplaces to have
+different methods of transfer a single token vs many 
+tokens. Other applications like wallets, will have
+a difficult time tracking consistently this information for many
+markets. This makes interactive with many NFT-driven apps 
+infeasible.
+
+Due to discussions here 
+https://github.com/near/NEPs/issues/254,
+a standard way to capture these events is needed.
+
+## Interface
+
+Many apps may use different mechanisms to perform their batching
+requirements. This interface standardizes this process by capturing
+events of these activities through logs.
+
+```ts
+// Interface to capture an event
+// and return formatted event string.
+interface EventJsonLog {
+    // Takes `EventLogData` and returns
+    // `EVENT_JSON: <EVENT_LOG_DATA>`
+    (e:EventLogData):string
+}
+
+// Interface to capture data 
+// about an event
+// Arguments
+// * `standard`: name of standard e.g. nep171
+// * `version`: e.g. 1.0.0
+// * `event`: `nft_mint` | `nft_burn` | `nft_transfer`
+// * `data`: associate event data
+interface EventLogData {
+    standard:string,
+    version:string,
+    event:string,
+    data: NftMintLog[]|NftTransferLog[]|NftBurnLog
+}
+
+// An event log to capture token minting
+// Arguments
+// * `account_id`: "account.near"
+// * `token_id`: "1"
+interface NftMintLog {
+    owner_id:string,
+    token_id:string
+}
+
+// An event log to capture token burning
+// Arguments
+// * `owner_id`: owner of tokens to burn
+// * `token_ids`: ["1","2"]
+interface NftBurnLog {
+    owner_id:string,
+    token_ids:string[]
```

Comment:
Are we sure we want to pass here the list? I think it's better to use here `token_id` as everywhere else not to mess up.

Reply:
I think in the majority of cases for batch minting and burning, these will be from the same user account and this makes sense. Similar to suggestions I'm seeing in the [multi-token standard](https://github.com/near/NEPs/pull/245).

Reply:
Do we want to pass the list of ids in mint also?

Not suggesting doing that for transfer, it sounds weird. But I didn't know the case with burn, so I could be wrong here.

Reply:
Good point, I definitely think we should do that with mint

Reply:
@evergreen-trading-system @telezhnaya @mikedotexe I believe that if we go with batching all the way down, we should bake this support everywhere: mint, transfer, burn.

```js
JSON_EVENT {"standard": "nep171", "version": "1.0.0", "event": "mint": "data": [
  {"owner_id": "frol.near", "token_ids": ["1", "2"]}
]}

JSON_EVENT {"standard": "nep171", "version": "1.0.0", "event": "transfer": "data": [
  {"sender_id": "frol.near", "receiver_id": "olga.near", "token_ids": ["1", "2"]}
]}

JSON_EVENT {"standard": "nep171", "version": "1.0.0", "event": "burn": "data": [
  {"owner_id": "frol.near", "approval_id": "olga.near", "token_ids": ["1", "2"]},
  {"owner_id": "frol.near", "token_ids": ["3"]}
]}
```

Reply:
@mikedotexe Do you have a good naming for "an account that was approved to perform transfer/burn operations on behalf of the NFT owner"? I currently named it as "representative_id" just for the sake of no better name I could come up with. I feel it is similar to "Intermediate / Corresponding Bank", but I am really bad in that area of terminology.